src/debugpy/server/cli.py

```diff
@@ -193,23 +191,71 @@ def do(arg, it):
 ]
 # fmt: on
 
-
+# Consume all the args from argv
 def consume_argv():
     while len(sys.argv) >= 2:
         value = sys.argv[1]
         del sys.argv[1]
         yield value
 
+# Consume all the args from a given list
+def consume_args(args: list):
+    if (args is sys.argv):
+        yield from consume_argv()
+    
+    while len(args) >= 1:
```

Comment:
Doesn't this go down this path after it yields the args from consume_argv()? Like isn't an else needed here?

Reply:
Yeah I think you are correct here. Although all my testing showed this as working, but maybe I missed an edge case. I will add the else for sure, since it seems more correct.

Reply:
This is fixed in the latest commit.